testonly/hammer/hammer.go

```diff
@@ -305,6 +305,10 @@ func (s *hammerState) pickKey(which int) []byte {
 	defer s.mu.RUnlock()
 
 	l := len(s.contents[which].data)
+	if l == 0 {
+		panic(fmt.Sprintf("internal error: can't pick a key, map contents[%d] is empty!", which))
```

Comment:
(pre-existing) Isn't possible to hit this if we're unlucky with the dice?  If we happen to do SetLeaves:CreateLeaf then SetLeaves:DeleteLeaf at start of day, we could end up with an empty map?

Reply:
If that happens, it's a bug in the hammer design.
This panic purely provides a useful message in the event that s.contents[which].data is empty.
The program would crash on the next line in any case since rand.Intn(0) itself panics.
I want to have a more useful message to help fix the bug if I see it happen.

Reply:
Yes, I think it is a (pre-existing) bug in the hammer design that hadn't occurred to me until now.  

(As such, there's obviously no need to fix it in this PR but it would be good if you could check whether you agree with me that it's theoretically possible.  If we had your shiny replay idea we would be able to test it easily by just replaying those two operations.)